### PR TITLE
Fixed configuration display issues, refactoring, preserving folder structure

### PR DESCRIPTION
**Pull request by r888800009**
* Fixed Zoom display inconsistency issue PR #10 
* Refactor source/nijigenerate/io/package.d confusing code
* Add the ~~KRA~~/PSD import "Preserve Folder Structure" configuration option (Currently only PSD works)
* Add ~~[message box](https://github.com/nijigenerate/nijigenerate/blob/64d80795156ac8b7b08e26f97b00d0d22e0773cd/source/nijigenerate/io/package.d#L164), [Import prompt window](https://github.com/nijigenerate/nijigenerate/blob/64d80795156ac8b7b08e26f97b00d0d22e0773cd/source/nijigenerate/io/package.d#L246)~~ DialogHandler

### CODE REVIEW

File: source/nijigenerate/windows/settings.d [hunk 1]
```diff
@@ -23,7 +23,7 @@ enum SettingsPane : string {
     LookAndFeel = "Look and Feel",
     Viewport = "Viewport",
     Accessibility = "Accessbility",
-    Autosaves = "Autosaves"
+    FileHandling = "FileHandling"
```
Comment (seagetch) [inline source/nijigenerate/windows/settings.d:26]: Maybe this should be "File Handling"

**Comment by r888800009** [comment]
@seagetch I marked the PR as ready for review, but I noticed that the KRA import code does not preserve the folder structure when the setting is enabled.

**Comment by r888800009** [comment]
screenshots
image
image
image

File: source/nijigenerate/io/package.d [hunk 2]
```diff
@@ -166,19 +153,35 @@ string incShowSaveDialog(const(TFD_Filter)[] filters, string fname, string title
...
     } else {
         c_str filename = tinyfd_saveFileDialog(title.toStringz, fname.toStringz, filters);
-        if (filename !is null) {
-            string file = cast(string) filename.fromStringz;
-            return file;
+        return incToDString(filename);
+    }
+}
+
+string incMessageBox(string title, string message, string dialogType = "ok", string iconType = "info", int defaultButton = 0) {
```
Comment (seagetch) [inline source/nijigenerate/io/package.d:164]: can you define enum instead of string?

e.g.

enum AskDialogAnswer {
   Cancel, Yes, No
}

**Comment by r888800009** [comment]
@seagetch `incMessageBox()` and `incImportKeepFolderStructPop()` has been changed, but the `tinyfiledialogs` library has poorly defined button IDs, so I need to implement a convert function.

**Comment by r888800009** [comment]
Check the `incDialog()` implementation in `source/nijigenerate/widgets/dialog.d` to see if it can replace `tinyfiledialogs`.

**Comment by r888800009** [comment]
I'm not sure which action is causing `incDialogButtonSelected()` to return `NONE`.

I might unifying the enum `DialogButton` with `DialogButtons` in `source/nijigenerate/widgets/dialog.d`.

```d
    incDialog(
        "Import_File_Ask", 
        __("Are you sure?"), 
        _("test message."),
        DialogLevel.Warning,
        DialogButtons.Yes | DialogButtons.No | DialogButtons.Cancel
    );

    writeln("incDialogButtonSelected: ", incDialogButtonSelected("Import_File_Ask"));
```

**Comment by seagetch** [comment]
let me know if you think PR is prepared for merge

**Comment by r888800009** [comment]
@seagetch Maybe this is sufficient? 
If switch to `incDialog` need to refactor most of `incDialog` to class or writing convoluted multi-layer callbacks. No elegant way to implement it. Maybe an event manager is needed first.

Currently, it might be possible to make the `incMessageBox` and `incDialog` interfaces consistent if necessary. Please let me know if you think this is necessary. If so, I will push a new commit.

By the way, based on the previous comment about `incDialogButtonSelected`. I'll mark below code as a potential issue, although it hasn't been verified yet.
https://github.com/nijigenerate/nijigenerate/blob/63902b04a4d2924efcc3a18494922e4ed5e481a6/source/nijigenerate/viewport/vertex/package.d#L168-L186

**Comment by r888800009** [comment]
@seagetch I'm currently implementing the class into `dialog.d` to avoid environment-related issues. However, I've noticed that `nijiui/source/nijiui/widgets` and `nijigenerate/source/nijigenerate/widgets` contain some similar code, and `nijigenerate` has no dependencies.

I'm not sure if there were specific reasons for not using `nijiui`. I'm considering whether I could directly use the `nijiui` program code instead.

https://github.com/nijigenerate/nijiui/tree/main/source/nijiui/widgets

https://github.com/nijigenerate/nijigenerate/tree/main/source/nijigenerate/widgets

**Comment by grillo-delmal** [comment]
> @seagetch I'm currently implementing the class into `dialog.d` to avoid environment-related issues. However, I've noticed that `nijiui/source/nijiui/widgets` and `nijigenerate/source/nijigenerate/widgets` contain some duplicate code, and `nijigenerate` has no dependencies.
> 
> I'm not sure if there were specific reasons for not using `nijiui`. I'm considering whether I could directly use the `nijiui` program code instead.
> 
> https://github.com/nijigenerate/nijiui/tree/main/source/nijiui/widgets
> 
> https://github.com/nijigenerate/nijigenerate/tree/main/source/nijigenerate/widgets

nijiui is a fork of inui, which it was made initially by Luna to try to separate the UI code from the program logic on both inochi-creator and inochi-session. She started doing this work in session, but ended up giving up along the way.

**Comment by r888800009** [comment]
@grillo-delmal This PR is not a large-scale refactoring of the project. It’s a minimum viable solution to ensure that users are not disappointed with the FOSS application.

While newer architectures may be developed in the future, they cannot address the current issues.

**Comment by r888800009** [comment]
@seagetch It now uses `dialog.d`, is no longer affected by the wrapper, and implements `DialogHandler` to manage click events.
 image

**Pull request merged**